eduNEXT/tutor-contrib-edunext-distro
# feat: add quince compatibility for distro

This PR adds quince compatibility for the distro plugin.

**Major changes**

1. Removed the "openedx-dev-dockerfile-post-python-requirements" patch since the private packages won't be installed in the extra_deps folder anymore.
2. The _enable-private packages_ command will run the tutor mounts add command for each private package and will write the right hook inside the plugin.py file so the private package will be built inisde the Docker file
3. Removed the exclusion rule towards the tests packages inside the setup.py file since the packages inside the tests modules weren't being imported correctly.
4. Changed minimum tutor version to 17.0.3 to include a fix to properly install private packages described here: https://github.com/overhangio/tutor/pull/1016

**How to test**

1. Install distro using this branch
2. enable distro using **tutor plugins enable distro**
5. Make sure your config file has the correct syntax to add themes and packages (include private packages) it can look something like this:

```
DISTRO_EOX_MANAGE_DPKG:
  domain: github.com
  index: git
  name: eox-manage
  path: eduNEXT
  private: true
  protocol: ssh
  repo: eox-manage
  variables:
    development: {}
    production: {}
  version: v5.0.0
DISTRO_EOX_SUPPORT_DPKG:
  domain: github.com
  index: git
  name: eox-support
  path: eduNEXT
  private: true
  protocol: ssh
  repo: eox-support
  variables:
    development: {}
    production: {}
  version: v7.0.0
DISTRO_EOX_THEMING_DPKG:
  domain: github.com
  index: git
  name: eox-theming
  path: eduNEXT
  private: false
  protocol: https
  repo: eox-theming
  variables:
    development:
      EOX_THEMING_CONFIG_SOURCES:
      - from_eox_tenant_microsite_v2
      - from_django_settings
    production:
      EOX_THEMING_CONFIG_SOURCES:
      - from_eox_tenant_microsite_v2
      - from_django_settings
  version: v7.0.0
DISTRO_EXTRA_MIDDLEWARES:
- eox_core.middleware.PathRedirectionMiddleware
- eox_core.middleware.RedirectionsMiddleware
DISTRO_THEMES:
- domain: github.com
  name: ednx-saas-themes
  path: eduNEXT
  protocol: ssh
  repo: ednx-saas-themes
  version: edunext/quince.master
DISTRO_THEMES_NAME:
- bragi
- css-runtime
DISTRO_THEMES_ROOT: /openedx/themes
DISTRO_THEME_DIRS:
- /openedx/themes/ednx-saas-themes/edx-platform
- /openedx/themes/ednx-saas-themes/edx-platform/bragi-generator
- /openedx/themes/ednx-saas-themes/edx-platform/bragi-children

OPENEDX_EXTRA_PIP_REQUIREMENTS:
- eox-tenant==v10.0.0
- eox-core==v10.0.0
- eox-hooks==v6.0.0
- eox-audit-model==v4.0.0
- eox-tagging==v7.0.0

```

6. Test the following commands :
```
tutor distro syntax-validator
tutor distro repository-validator
tutor distro enable-themes
tutor distro enable-private-packages
```
7. Run tutor build images openedx and check if there was no error
8. Launch your environment and inside your container run pip list and verify that all the packages are installed
